src/TrackPerfHistProc_NoTracking.cxx

```diff
@@ -0,0 +1,325 @@
+			   _ibtrkhitColName
+			   );  
+
+  registerInputCollection( LCIO::TRACKERHIT,
+			   "OBTrackerHitsCollection" , 
+			   "Name of outer barrel tracker hits collection",
+			   _obtrkhitColName,
+			   _obtrkhitColName
+			   );  
+
+  registerInputCollection( LCIO::TRACKERHIT,
+			   "VETrackerHitsCollection" , 
+			   "Name of vertex endcap tracker hits collection",
+			   _vetrkhitColName,
+			   _vetrkhitColName
+			   );  
+
+  registerInputCollection( LCIO::TRACKERHIT,
+			   "IETrackerHitsCollection" , 
+			   "Name of inner endcap tracker hits collection",
+			   _ietrkhitColName,
+			   _ietrkhitColName
+			   );     
+
+  registerInputCollection( LCIO::TRACKERHIT,
+			   "OETrackerHitsCollection" , 
+			   "Name of outer endcap tracker hits collection",
+			   _oetrkhitColName,
+			   _oetrkhitColName
+			   );  
+
+  registerInputCollection( LCIO::LCRELATION,
+		     "VBRelationCollection" ,
+			   "Name of the input relation collection",
+			   _VBRelationCollection,
+		     _VBRelationCollection
+		 	    );
+}
+
+void TrackPerfHistProc::init()
+{
+  // Print the initial parameters
+  printParameters() ;
+
+  // Create histograms
+  AIDA::ITree* tree=marlin::AIDAProcessor::tree(this);
+  marlin::AIDAProcessor::histogramFactory(this);
+
+  // Create ROOT histograms, with location setup by the above factory
+  tree->mkdir("all" ); tree->cd("all" );
+  _allTracks =std::make_shared<TrackPerf::TrackHists>();
+  _allTruths =std::make_shared<TrackPerf::TruthHists>();
+  h_number_of_tracks = new TH1F("number_of_tracks", ";Number of seeds/tracks;Events", 100, 0, 300000);
+  h_trackerhit_timing = new TH1F("time_of_flight", ";Time of flight for tracker hits [ns];Tracker hits;", 1000, -1, 1);
+  tree->mkdir("../real"); tree->cd("../real");
+  _realTracks=std::make_shared<TrackPerf::TrackHists>();
+  _realTruths=std::make_shared<TrackPerf::TruthHists>();
+  _realReso    =std::make_shared<TrackPerf::TrackResoHists>();
+  h_relation_weight_real = new TH1F("relation_weight", ";Track-truth relation weight;", 100, 0, 2);
+  tree->mkdir("../fake"); tree->cd("../fake");
+  _fakeTracks=std::make_shared<TrackPerf::TrackHists>();
+  h_number_of_fakes = new TH1F("number_of_fakes", ";Number of fake tracks;Events", 100,  0, 300000);
+  h_relation_weight_fake = new TH1F("relation_weight", ";Track-truth relation weight;Tracks;", 100, 0, 2);
+  tree->mkdir("../unmt"); tree->cd("../unmt");
+  _unmtTruths=std::make_shared<TrackPerf::TruthHists>(); 
+
+  tree->mkdir("../clusters" ); tree->cd("../clusters" );
+  _uncertainties=std::make_shared<TrackPerf::TrackerHitResoHists>();
+  _clusters=std::make_shared<TrackPerf::ClusterHists>();
+
+}
+
+void TrackPerfHistProc::processRunHeader( LCRunHeader* /*run*/)
+{ } 
+
+void TrackPerfHistProc::processEvent( LCEvent * evt )
+{
+  //
+  // Get object required collections and create lists
+  // to keep track of unsaved objects.
+
+  // MCParticles
+
+  /* LCCollection* mcpCol  =evt->getCollection(_mcpColName);
```

Comment:
Instead of making a new processor, it would be good if here you simply check if the getCollection returns a valid pointer and, if not, you skip the tracking validation part and go to the cluster one (and do a similar check there).
Making a new processor duplicates lots of code for no good reason.